src/main/java/org/javamodularity/moduleplugin/JavaProjectHelper.java

```diff
@@ -52,6 +56,40 @@ public SourceSet mainSourceSet() {
     public SourceSet testSourceSet() {
         return sourceSet(SourceSet.TEST_SOURCE_SET_NAME);
     }
+
+    public GroovySourceSet groovySourceSet(SourceSet javaSourceSet) {
```

Comment:
I don't like this implementation because it uses DslObject, which is internal API, but it seems there is no other way to do it (except by using reflection).

Reply:
Yes, I was wondering if the impl can be improved. I gave it a look based on [how Kotlin does it](https://kotlinlang.org/docs/reference/using-gradle.html#targeting-the-jvm), and came up with the following piece of code:
```java
    public GroovySourceSet groovySourceSet(SourceSet sourceSet) {
        return ((HasConvention) sourceSet).getConvention().getPlugin(GroovySourceSet.class);
    }
```

Would you give it a try?